### PR TITLE
PT-2259: Update Pricing module

### PR DESCRIPTION
**Pull request by aizhuli**
## Description
Implement generic CRUD service

### CODE REVIEW

File: src/VirtoCommerce.PricingModule.Core/Services/IPriceService.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using VirtoCommerce.PricingModule.Core.Model;
+
+namespace VirtoCommerce.PricingModule.Core.Services
+{
+    public interface IPriceService
```
Comment (eMazeika) [inline src/VirtoCommerce.PricingModule.Core/Services/IPriceService.cs:7]: IPriceService : ICrudService<Price>
was there any reason to skip this? Then the conversion could be removed in all the code

File: src/VirtoCommerce.PricingModule.Data/Handlers/DeletePricesProductChangedEventHandler.cs [hunk 2]
```diff
@@ -4,20 +4,22 @@
...
 using VirtoCommerce.PricingModule.Core.Model.Search;
 using VirtoCommerce.PricingModule.Core.Services;
 
 namespace VirtoCommerce.PricingModule.Data.Handlers
 {
     public class DeletePricesProductChangedEventHandler : IEventHandler<ProductChangedEvent>
     {
-        private readonly IPricingService _pricingService;
-        private readonly IPricingSearchService _pricingSearchService;
+        private readonly ICrudService<Price> _priceService;
```
Comment (eMazeika) [inline src/VirtoCommerce.PricingModule.Data/Handlers/DeletePricesProductChangedEventHandler.cs:16]: IPriceService _priceService
Confusing to see the same name of different types. The _priceService variable should be of the same type in all the module code.

File: src/VirtoCommerce.PricingModule.Core/Services/IPriceSearchService.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+namespace VirtoCommerce.PricingModule.Core.Services
+{
+    public interface IPriceSearchService
```
Comment (eMazeika) [inline src/VirtoCommerce.PricingModule.Core/Services/IPriceSearchService.cs:3]: : ISearchService<PricesSearchCriteria, PriceSearchResult, Price>

File: src/VirtoCommerce.PricingModule.Core/Services/IPriceService.cs (hunk 1 above)
Comment (aizhuli) [reply src/VirtoCommerce.PricingModule.Core/Services/IPriceService.cs:7]: I am not sure I understand the comment, this is a new interface, PriceService class implements this interface along with the CrudService<..>...

File: src/VirtoCommerce.PricingModule.Core/Services/IPriceSearchService.cs (hunk 3 above)
Comment (aizhuli) [reply src/VirtoCommerce.PricingModule.Core/Services/IPriceSearchService.cs:3]: PriceSearchService class implements this interface and SearchService<PricesSearchCriteria, PriceSearchResult, Price>...

**Comment by akak1977** [comment]
New PR with minimal changes and preserved compatibility:
https://github.com/VirtoCommerce/vc-module-pricing/pull/151

**Pull request closed without merging**